src/Microsoft.HttpRepl.Telemetry/Telemetry.cs

```diff
@@ -0,0 +1,202 @@
+            IReadOnlyDictionary<string, double> measurements)
+        {
+            if (!Enabled)
+            {
+                return;
+            }
+
+            //continue task in existing parallel thread
+            _trackEventTask = _trackEventTask.ContinueWith(
+                x => TrackEventTask(eventName, properties, measurements),
+                TaskScheduler.Default
+            );
+        }
+
+        private void ThreadBlockingTrackEvent(string eventName, IReadOnlyDictionary<string, string> properties, IReadOnlyDictionary<string, double> measurements)
+        {
+            if (!Enabled)
+            {
+                return;
+            }
+            TrackEventTask(eventName, properties, measurements);
+        }
+
+        private void InitializeTelemetry(string productVersion)
+        {
+            try
+            {
+#pragma warning disable CS0618 // Type or member is obsolete
+                _client = new TelemetryClient();
+#pragma warning restore CS0618 // Type or member is obsolete
+                _client.InstrumentationKey = InstrumentationKey;
+                _client.Context.Session.Id = CurrentSessionId;
+                _client.Context.Device.OperatingSystem = RuntimeEnvironment.OperatingSystem;
+
+                _commonProperties = new TelemetryCommonProperties(productVersion).GetTelemetryCommonProperties();
+                _commonMeasurements = new Dictionary<string, double>();
+            }
+            catch (Exception e)
+            {
+                _client = null;
+                // we dont want to fail the tool if telemetry fails.
+                Debug.Fail(e.ToString());
+            }
+        }
+
+        private void TrackEventTask(
+            string eventName,
+            IReadOnlyDictionary<string, string> properties,
+            IReadOnlyDictionary<string, double> measurements)
+        {
+            if (_client == null)
+            {
+                return;
+            }
+
+            try
+            {
+                Dictionary<string, string> eventProperties = GetEventProperties(properties);
+                Dictionary<string, double> eventMeasurements = GetEventMeasures(measurements);
+
+                _client.TrackEvent(PrependProducerNamespace(eventName), eventProperties, eventMeasurements);
+                _client.Flush();
+            }
+            catch (Exception e)
+            {
+                Debug.Fail(e.ToString());
+            }
+        }
+
+        private static string PrependProducerNamespace(string eventName)
+        {
+            return "dotnet/httprepl/" + eventName;
+        }
+
+        private Dictionary<string, double> GetEventMeasures(IReadOnlyDictionary<string, double> measurements)
+        {
+            Dictionary<string, double> eventMeasurements = new Dictionary<string, double>(_commonMeasurements);
+            if (measurements != null)
```

Comment:
I don't know. Interestingly, both were implemented the same way originally, then only GetEventProperties was changed and not this one - and the commit says it was in response to PR feedback. I can't seem to find the PR though, so I'm hesitant to change it even though I don't why it would matter.